code_docs/Organization.md

```diff
@@ -44,6 +44,21 @@ The documentation structure would look something like this:
 
 
 ### Code Cleanup
+#### Automation View Refactoring
+**Proposal:**
+When refactoring it would make sense to create a class `AutomationEditor` which just implements the editor itself. Ideally this class would implement a few different layouts for the editing interface, for example the velocity editing layout. When launching the editor, the layout can be set appropriately to the parameter edited.
+
+If necessary, subclasses could be created to implement different behavior. But as all the editors look extremely similar, I feel like having them all in one class would be most efficient.
+
+A separate class `AutomationEditorLaunchpad` would implement the launchpad.
+
+`AutomationEditorLaunchpad` presents different sets of parameters to edit depending on where/how it is launched. Instrument Clip View => [audition pad] + [song] could show Note Automation.
+
+Considerations:
+- Note automation like probability could be automated on clip *and* note level. (Which parameters should be automatable and how is a different discussion.)
+- Some parameters will probably need a new property to be saved in, which the `AutomationEditor` will translate to a note/synth parameter depending on its editing layout
+- The shortcut for calling the Launch Pad doesn't feel very consistent currently as it's either on the song or clip button. Ideally this would live on the same button wherever it can be called
```

Comment:
So we have two shortcuts:

- clip in clip views. pressing the clip button to enter the launch pad / overview was selected because it felt analogous to switching between song view and arranger view. I like how this feels.
- using the clip button didn't make sense while in arranger view so we opted for shift + song

You can also enter automation editor via the menu as well. With a parameter menu open, pressing song while in the arranger menu opens the automation editor for the arranger parameter and pressing clip while in the. clip menu opens the automation editor for the clip parameter.

Reply:
The current shortcuts definitely aren't bad choices! But with automation editing becoming larger and more important in sequencing, I feel it would be great to have a single spot for you muscle memory to move towards, without having to check first what view you're in

Reply:
> The current shortcuts definitely aren't bad choices! But with automation editing becoming larger and more important in sequencing, I feel it would be great to have a single spot for you muscle memory to move towards, without having to check first what view you're in
> 
> **Edit:** I'm trying out some button combinations and I guess I just feel differently about this, especially as the clip button is currently unused in arranger mode. This allows for a single clip press to open arranger automation, and holding a clip + clip button to directly open clip automation. The consistency between interfaces, of result of a button-press, makes it make sense to me

Oh yea just checked and you're right the clip button isn't used. We could definitely move the shortcut to clip then for consistency.